tests/fixtures/kafka.py

```diff
@@ -0,0 +1,97 @@
+  LOCAL = {
+    kdc = kdc.marathon.autoip.dcos.thisdcos.directory:2500
+  }
+'''
+KAFKA_KRB5 = base64.b64encode(KAFKA_KRB5_ORIG).decode('utf8')
+KAFKA_PACKAGE_NAME = os.getenv("KAFKA_PACKAGE_NAME", "kafka")
+KAFKA_SERVICE_NAME = os.getenv("KAFKA_SERVICE_NAME", ("secure-kafka" if KERBERIZED_KAFKA else "kafka"))
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+
+KEYTAB_SECRET = "__dcos_base64___keytab"
+
+def upload_jaas():
+    jaas_path = os.path.join(THIS_DIR, "..", "resources", "spark-kafka-client-jaas.conf")
+    s3.upload_file(jaas_path)
+    return s3.http_url("spark-kafka-client-jaas.conf")
+
+
+def get_kerberized_kafka_spark_conf(spark_service_name, keytab_secret=KEYTAB_SECRET):
+    return [
+        "--conf", "spark.mesos.driver.secret.names={}".format(keytab_secret),
+        "--conf", "spark.mesos.driver.secret.filenames=kafka-client.keytab",
+        "--conf", "spark.mesos.executor.secret.names={}".format(keytab_secret),
+        "--conf", "spark.mesos.executor.secret.filenames=kafka-client.keytab",
+        "--conf", "spark.mesos.task.labels=DCOS_SPACE:/{}".format(spark_service_name),
+        "--conf", "spark.executorEnv.KRB5_CONFIG_BASE64={}".format(KAFKA_KRB5),
+        "--conf", "spark.mesos.driverEnv.KRB5_CONFIG_BASE64={}".format(KAFKA_KRB5),
+        "--conf", "spark.driver.extraJavaOptions=-Djava.security.auth.login.config="
+                  "/mnt/mesos/sandbox/spark-kafka-client-jaas.conf",
+        "--conf", "spark.executor.extraJavaOptions="
+                  "-Djava.security.auth.login.config=/mnt/mesos/sandbox/spark-kafka-client-jaas.conf",
+    ]
+
+
+@pytest.fixture(scope='session')
+def configure_security_kafka():
+    yield from sdk_security.security_session(KAFKA_SERVICE_NAME)
+
+
+@pytest.fixture(scope='session')
+def kerberized_kafka(configure_security_kafka, kerberos_env):
+    try:
+        kerberos_options = {
+            "kerberos": {
+                "enabled": True,
+                "kdc": {
+                    "hostname": kerberos_env.get_host(),
+                    "port": int(kerberos_env.get_port())
+                },
+                "keytab_secret": kerberos_env.get_keytab_path(),
+                "realm": kerberos_env.get_realm()
+            }
+        }
+
+        service_options = {
+            "service": {
+                "name": KAFKA_SERVICE_NAME,
+                "security": kerberos_options
+            },
+            "kafka": {
+                "default_replication_factor": 3,
+                "num_partitions": 32
+            }
+        }
+
+        sdk_install.uninstall(KAFKA_PACKAGE_NAME, KAFKA_SERVICE_NAME)
+        sdk_install.install(
+            KAFKA_PACKAGE_NAME,
+            KAFKA_SERVICE_NAME,
+            DEFAULT_KAFKA_TASK_COUNT,
+            additional_options=service_options,
+            timeout_seconds=30 * 60)
+
+        yield kerberos_env
+
+    finally:
+        sdk_install.uninstall(KAFKA_PACKAGE_NAME, KAFKA_SERVICE_NAME)
+        kerberos_env.cleanup()
```

Comment:
As is the case with HDFS, we should not cleanup the Kerberos environment here.

Reply:
cleanup moved to KDC fixture